pkg/disk/x509.go

```diff
@@ -0,0 +1,96 @@
+package disk
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+)
+
+const (
+	certsFileMode = os.FileMode(0644)
+	keyFileMode   = os.FileMode(0600)
+)
+
+// WriteX509Context takes a X509Context, representing a svid message from
+// the Workload API, and calls writeCerts and writeKey to write to disk
+// the svid, key and bundle of certificates.
+// It is possible to change output setting `addIntermediatesToBundle` as true.
+func WriteX509Context(x509Context *workloadapi.X509Context, addIntermediatesToBundle, includeFederatedDomains bool, certDir, svidFilename, svidKeyFilename, svidBundleFilename string) error {
+	svidFile := path.Join(certDir, svidFilename)
+	svidKeyFile := path.Join(certDir, svidKeyFilename)
+	svidBundleFile := path.Join(certDir, svidBundleFilename)
+
+	// There may be more than one certificate, but we're only interested in the default one
+	svid := x509Context.DefaultSVID()
+	certs := svid.Certificates
```

Comment:
ca nyou move certs definition near to addIntermediatesToBundle, first time it is used, to simplify code view?

Reply:
done